baby-gru/public/baby-gru/CootWorker.ts

```diff
@@ -911,6 +912,37 @@ const setUserDefinedBondColours = (imol: number, colours: { cid: string; rgb: [n
     colourMap.delete()
 }
 
+const run_privateer_validation = (fileContents: string, fileName: string) => { 
+    return cootModule.validate(fileContents, fileName)
+}
+
+const extract_carbohydrate_validation = (results: any) : PrivateerResultsEntry[] => {
+
+    const sanitizeID = (id: string): string => {
+        const regex = /: *32/g;
+        return id.replace(regex, '');
+    }
+
+    const data: PrivateerResultsEntry[] = [];
+    for (let i = 0; i < results.size(); i++) {
+        const entry = results.get(i);
+
+        entry.id = sanitizeID(entry.id as string);
+
+        const collectedTorsions: any[] = [];
+        for (let j = 0; j < entry.torsions.size(); j++) {
+            collectedTorsions.push(entry.torsions.get(j));
+        }
+        entry.torsions = collectedTorsions;
+        const regex = /: *32/g;
+        entry.svg = entry.svg.replace(regex, '');
+        data.push(entry as PrivateerResultsEntry);
```

Comment:
I would feel more comfortable if you created a new object `entryJS` (example name) and explicitly fill it with the data that we want to export to JS instead of replacing the fields of `entry` in situ. That way we make sure we don't pass C++ handles.